### ISSUE TITLE
Make the collision detection work for all kind of detection

### ISSUE THREAD

**Issue by Lu-ni** [jjballlz/sdlGame#7]
Actually the collision detection is between man and ledges. In the future we are going to need a lot of collision detection for spell, monster and other.

**Comment by jjballlz** [comment]
yup was thinking of attacking this this evening, you worked on physics.h since last push?
